packages/manager/cypress/e2e/core/cloudpulse/widget-verification.spec.ts

```
@@ -0,0 +1,103 @@
+    linodeWidgets.forEach((testData) => {
+      verifyGranularity(testData.title, testData.expectedGranularityArray);
+    });
+  });
+
+  it(`should verify available aggregation  of the widget`, () => {
+    linodeWidgets.forEach((testData) => {
+      verifyAggregation(testData.title, testData.expectedAggregationArray);
+    });
+  });
+  it(`should zoom in and out of the all the widget`, () => {
+    linodeWidgets.forEach((testData) => {
+      checkZoomActions(testData.title);
+    });
+  });
+  it('should apply global refresh button and verify network calls', () => {
+    performGlobalRefresh();
+    interceptMetricsRequests().then((xhrArray) => {
+      xhrArray.forEach((xhr) => {
+        const { body: requestPayload } = xhr.request;
+        const metric = requestPayload.metric;
+        const metricData = linodeWidgets.find(
+          (data) => data.name === metric
+        );
+      if (!metricData) {
+          throw new Error(`Unknown metric: ${metric}`);
+        }
+        const granularity = requestPayload['time_granularity'];
+        const currentGranularity = granularity
+          ? granularity.value + granularity.unit
+          : '';
+        const durationUnit = requestPayload.relative_time_duration.unit.toLowerCase();
+        const durationValue = requestPayload.relative_time_duration.value;
+        const currentRelativeTimeDuration =
+          durationUnit in timeUnit
+            ? 'Last' +
+              durationValue +
+              timeUnit[durationUnit as keyof typeof timeUnit]
+            : '';
+        expect(requestPayload.aggregate_function).to.equal(
+          metricData.expectedAggregation
+        );
+        expect(currentRelativeTimeDuration).to.containIgnoreSpaces(
+          actualRelativeTimeDuration
+        );
+        expect(currentGranularity).to.containIgnoreSpaces(
+          metricData.expectedGranularity
+        );
+        expect(requestPayload.metric).to.containIgnoreSpaces(metricData.name);
```

We're on the right track with this test and I like what we're testing!

I think there's some opportunity to clean this up a bit, though:

- Let's use template literals for concatenating strings, like \`${granularity.value}${granularity.unit}\` instead of `granularity.value + granularity.unit`
    - I've noticed this done a few places throughout this PR; let's clean all of them up
- What's going on with the `timeUnit[durationUnit as keyof typeof timeUnit]`?
- The ternary on line 83 is kind of weird -- I'm having trouble understanding what condition you're trying to check for, but if it doesn't meet that condition let's throw an error and fail the test instead of assigning an empty string (kinda like what you did up on line 74)
- Why do we have to use `containIgnoreSpaces`? Shouldn't we be able to test for the exact value?